hazelcast/src/test/java/com/hazelcast/internal/util/JavaVersionTest.java

```diff
@@ -64,4 +71,40 @@ public void testIsAtMost() {
             assertTrue(JavaVersion.isAtMost(thisVersion, new FutureJavaVersion(thisVersion.getMajorVersion())));
         }
     }
+
+    @Test
+    public void testVersionDetection() {
+        String version = System.getProperty("java.version");
```

Comment:
We have a custom JUnit rule for this - `OverridePropertyRule`. It'll return the value to the proper one even if the test fails.

And please change the runner in the `@RunWith` annotation from the parallel one to the serial, it'll be safer due to system property touches.